packages/store/src/StoreCore.sol

```diff
@@ -213,12 +217,17 @@ library StoreCore {
     if (ResourceIds._getExists(tableId)) {
       revert IStoreErrors.Store_TableAlreadyExists(tableId, string(abi.encodePacked(tableId)));
     }
```

Comment:
Rather than adding a new table, I think we can just extend this check to look for `ResourceId` with the same name but different type.

Reply:
How else would you find a `ResourceId` with the same name? There is no global array of all Resource IDs we can traverse

Reply:
Like, just given a `bytes30` name, how do you find any/all corresponding Resource ID's unless there is a table that specifically indexes that

Reply:
I see, we can change the type of Table ID to check if there is either an offchain or regular table instead